server.R

```diff
@@ -50,17 +83,37 @@ server <- function(input, output,session) {
     if(input$subtype == "Multiplot" && length(input$genes) > 0) {
       query <- geneQuery(genes = input$genes, table = input$dataset)
       tcga <- dbGetQuery(database,query)
-
       query <- clinicalQuery(factors = "*", dataset = input$dataset)
       clinical <- dbGetQuery(database,query)
       clinical <- merge(tcga, clinical, by="UPN")
 
-      g <- ggplot(clinical,aes(fill=Gene, y=Expression, x=UPN, text = paste0("UPN ID: ",UPN,"<br />Dataset: ", input$dataset))) + geom_bar(position="dodge", stat="identity") + theme_bw() +
-        theme(text=element_text(size=12, family="avenir", face="bold"), axis.text=element_text(size=10, family="avenir", face="bold"),
-              axis.title=element_text(size=12, family="avenir", face="bold"),
-              axis.text.x = element_text(angle = 90, hjust = 1)) +
-        ggtitle("Multiple gene view") +
-        ylab("Log2 Expression") + xlab("")
+      # Render log2/raw graph based on user input
+      if (show.raw()){
+        g <- ggplot(clinical,aes(fill=Gene,
+                                 y=2^Expression,
```

Comment:
Would it make the code simpler to scale the values before the plotting code. So instead of two plotting chunks, you would do something like: 
```
if(show.log2){
   clinical$Expression = log2(clinical$Expression)
} 
```
then just one complicated `ggplot` block below. 
Is there a downside?  (I think you're already keeping track of the state of the data with the show.raw/show.log2 param so you'll know when and which way to toggle, right?)

Reply:
this line actually modifies the original data in the plot. St like this maybe better:
"""
if (show.log2) g <- ggplot(y=Expression)
else : g <- ggplot(y=2^expression)

g + big_data_stuff
"""

Reply:
Sure - that's eminently reasonable